src/schema/rules/checks/eeg.yaml

```diff
@@ -0,0 +1,31 @@
+---
+EEGChannelCountReq:
+  selectors:
+    - suffix == "eeg"
+  checks:
+    - sidecar.EEGChannelCount == count(associations.channels.type, "EEG")
+ECGChannelCountReq:
+  selectors:
+    - suffix == "eeg"
+  checks:
+    - sidecar.ECGChannelCount == count(associations.channels.type, "ECG")
+EMGChannelCountReq:
+  selectors:
+    - suffix == "eeg"
+  checks:
+    - sidecar.EMGChannelCount == count(associations.channels.type, "EMG")
+EOGChannelCountReq:
+  selectors:
+    - suffix == "eeg"
+  checks:
+    - sidecar.EOGChannelCount == count(associations.channels.type, "EOG")
+MiscChannelCountReq:
+  selectors:
+    - suffix == "eeg"
+  checks:
+    - sidecar.MiscChannelCount == count(associations.channels.type, "MISC")
+TriggerChannelCountReq:
+  selectors:
+    - suffix == "eeg"
+  checks:
+    - sidecar.TriggerChannelCount == count(associations.channels.type, "TRIG")
```

Comment:
We should make sure there's an associated channels file. We can do that by trying to access it:

```suggestion
    - suffix == "eeg"
    - associations.channels.type
  checks:
    - sidecar.EEGChannelCount == count(associations.channels.type, "EEG")
ECGChannelCountReq:
  selectors:
    - suffix == "eeg"
    - associations.channels.type
  checks:
    - sidecar.ECGChannelCount == count(associations.channels.type, "ECG")
EMGChannelCountReq:
  selectors:
    - suffix == "eeg"
    - associations.channels.type
  checks:
    - sidecar.EMGChannelCount == count(associations.channels.type, "EMG")
EOGChannelCountReq:
  selectors:
    - suffix == "eeg"
    - associations.channels.type
  checks:
    - sidecar.EOGChannelCount == count(associations.channels.type, "EOG")
MiscChannelCountReq:
  selectors:
    - suffix == "eeg"
    - associations.channels.type
  checks:
    - sidecar.MiscChannelCount == count(associations.channels.type, "MISC")
TriggerChannelCountReq:
  selectors:
    - suffix == "eeg"
    - associations.channels.type
  checks:
    - sidecar.TriggerChannelCount == count(associations.channels.type, "TRIG")
```

Is it possible for any of these sidecar fields to be missing? For example, if there are no TRIG channels, will `TriggerChannelCount` be 0 or absent?

Reply:
Thanks for the suggestion!

> Is it possible for any of these sidecar fields to be missing? For example, if there are no TRIG channels, will TriggerChannelCount be 0 or absent?

All `*ChannelCount` variables are RECOMMENDED, so they may be absent. This check is intended to be more like: "If you have the channels file AND the appropriate sidecar with the metadata, ... do they make sense together?" I would not warn if a `*ChannelCount` metadata item is absent, only if it is present AND does not correspond to the channels file.

Reply:
Got it. Updated the suggestion with a check that the field is defined. You'll also need to add an `issue` to each of these, e.g.,

```yaml
TriggerChannelCountReq:
  issue:
    code: TRIGGER_CHANNEL_COUNT_MISMATCH
    message: |
      The TriggerChannelCount metadata does not match the number of channels
      with type TRIG in the associated channels.tsv file.
    level: warning
```

Reply:
> Got it. Updated the suggestion with a check that the field is defined. You'll also need to add an issue to each of these, e.g.,

done -- note that this isn't present in the "template" I used for this PR: https://github.com/bids-standard/bids-specification/blob/eba3e08e2dbfd3ad2e57c2a52c3bda8c1961dc6c/src/schema/rules/checks/nirs.yaml#L23-L28